`magic-string` v1 `SourceMap` is not assignable to rolldown's `SourceMapInput` under `exactOptionalPropertyTypes`

### Reproduction link or steps

https://github.com/issueset/rolldown-magic-string-v1-repro.git

A Vite/Rolldown plugin that returns `{ code, map: ms.generateMap() }` from a `transform` hook fails to typecheck with `magic-string` v1 when the consuming project enables `exactOptionalPropertyTypes` in its `tsconfig.json`. The same plugin typechecks fine with `magic-string` v0.30.

### Steps to reproduce

```sh
pnpm install
pnpm typecheck
```

This repo contains two identical minimal Vite plugins:

- `src/plugin-magic-string-v0.ts` uses `magic-string@0.30.21` (aliased as `magic-string-v0`): typechecks fine.
- `src/plugin-magic-string-v1.ts` uses `magic-string@1.0.0` (aliased as `magic-string-v1`): fails to typecheck.


### What is expected?

No error 

### What is actually happening?

```
src/plugin-magic-string-v1.ts(12,3): error TS2322: Type '(this: TransformPluginContext, code: string) => { code: string; map: SourceMap; }' is not assignable to type 'ObjectHook<...>'.
  ...
        Type '{ code: string; map: SourceMap; }' is not assignable to type 'Omit<SourceDescription, "code">'.
          Types of property 'map' are incompatible.
            Type 'SourceMap' is not assignable to type 'SourceMapInput'.
              Type 'SourceMap' is not assignable to type 'ExistingRawSourceMap' with 'exactOptionalPropertyTypes: true'. Consider adding 'undefined' to the types of the target's properties.
                Types of property 'file' are incompatible.
                  Type 'string | undefined' is not assignable to type 'string | null'.
                    Type 'undefined' is not assignable to type 'string | null'.
```

### System Info

```Shell
System:
    OS: macOS 26.5.2
    CPU: (8) arm64 Apple M2
    Memory: 140.61 MB / 24.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 24.15.0 - /opt/homebrew/opt/node@24/bin/node
    Yarn: 1.22.22 - /opt/homebrew/opt/node@24/bin/yarn
    npm: 11.12.1 - /opt/homebrew/opt/node@24/bin/npm
    pnpm: 10.28.1 - /opt/homebrew/opt/node@24/bin/pnpm
    bun: 1.3.6 - /opt/homebrew/bin/bun
    Deno: 2.6.10 - /opt/homebrew/bin/deno
  Browsers:
    Chrome: 150.0.7871.128
    Chrome Canary: 152.0.7956.0
    Firefox: 152.0.6
    Firefox Developer Edition: 148.0
    Safari: 26.5.2
    Safari Technology Preview: 26.0
```

### Any additional comments?

### root cause


`magic-string` v1 changed the declared type of `SourceMap` properties from optional (`file?: string`) or required (`file: string`) to required-but-possibly-undefined:

```ts
// magic-string@0.30.21 
// npm:    https://npmx.dev/package-code/magic-string/v/0.30.21/dist%2Fmagic-string.es.d.mts#L46
// source: https://github.com/Rich-Harris/magic-string/blob/v0.30.21/src/index.d.ts#L46-L55
export class SourceMap {
	file: string;
	sourcesContent?: string[];
	x_google_ignoreList?: number[];
	// ...
}

// magic-string@1.0.0 
// npm:    https://npmx.dev/package-code/magic-string/v/1.0.0/dist%2Findex.d.mts#L37
// source: https://github.com/Rich-Harris/magic-string/blob/v1.0.0/src/SourceMap.ts#L70-L77
declare class SourceMap {
	file: string | undefined;
	sourcesContent: Array<string | null> | undefined;
	x_google_ignoreList: number[] | undefined;
	// ...
}
```

Rolldown's `ExistingRawSourceMap` declares these properties as optional without `undefined` in the type:

```ts
// rolldown@1.1.5 
// npm:    https://npmx.dev/package-code/rolldown/v/1.1.5/dist%2Fshared%2Fdefine-config-BhJ90aEv.d.mts#L2555
// source: https://github.com/rolldown/rolldown/blob/v1.1.5/packages/rolldown/src/types/sourcemap.ts#L5-L14
interface ExistingRawSourceMap {
	file?: string | null;
	mappings: string;
	names?: string[];
	sources?: (string | null)[];
	sourcesContent?: (string | null | undefined)[];
	sourceRoot?: string;
	version?: number;
	x_google_ignoreList?: number[];
}
```

With `exactOptionalPropertyTypes: true`, `file?: string | null` means the property may be absent, but if present it must not be `undefined`. So `magic-string` v1's `file: string | undefined` is not assignable, and the same applies to `sourcesContent` and `x_google_ignoreList`.

--- a/packages/rolldown/src/types/sourcemap.ts
+++ b/packages/rolldown/src/types/sourcemap.ts
@@ -3,14 +3,14 @@ import type { SourceMap } from './rolldown-output';
 
 /** @category Plugin APIs */
 export interface ExistingRawSourceMap {
-  file?: string | null;
+  file?: string | null | undefined;
   mappings: string;
-  names?: string[];
-  sources?: (string | null)[];
-  sourcesContent?: (string | null | undefined)[];
-  sourceRoot?: string;
-  version?: number; // make it optional to compat { mappings: '' }
-  x_google_ignoreList?: number[];
+  names?: string[] | undefined;
+  sources?: (string | null)[] | undefined;
+  sourcesContent?: (string | null | undefined)[] | undefined;
+  sourceRoot?: string | undefined;
+  version?: number | undefined; // make it optional to compat { mappings: '' }
+  x_google_ignoreList?: number[] | undefined;
 }
 
 /** @inline @category Plugin APIs */

